crates/ra_ide/src/inlay_hints.rs

```diff
@@ -245,8 +256,14 @@ fn should_show_param_hint(
     } else {
         fn_signature.parameters.len()
     };
+
     // avoid displaying hints for common functions like map, filter, etc.
-    if parameters_len == 1 && (param_name.len() == 1 || param_name == "predicate") {
+    // or other obvious words used in std
+    let is_obvious_param_name = match param_name {
+        "predicate" | "value" | "pat" | "rhs" | "other" => true,
+        _ => false,
+    };
+    if parameters_len == 1 && (param_name.len() == 1 || is_obvious_param_name) {
```

Comment:
Let's extract the `(param_name.len() == 1 || is_obvious_param_name)` into a 

```
fn is_obvious_param(param: &str) -> bool
```

function?